### DISCUSSION TITLE
Disable Some Fields for Downloadable Product

### DISCUSSION THREAD

**Discussion by balkisanton** [woocommerce/woocommerce#49617] [Checkout Flow]
I'm new using WooCommerce, I want to disable address, city, state, and zip code field for minimalism and faster checkout. But no single code work, even the custom plugin that worked before

why not just add an option to show or remove it in wordpress gutenberg block similar to removing cell phone number field? (screenshot attached)

![request](https://github.com/user-attachments/assets/d7ad9532-e25f-44c6-849a-64c7fc8a7948)

**Comment by balkisanton** [comment]
Please add that feature, thank you

**Comment by PlatonB** [comment]
The [discussion on this subject](https://github.com/woocommerce/woocommerce/discussions/42996#discussioncomment-7974437), unfortunately, has led nowhere.

**Comment by mikejolley** [comment]
This comment https://github.com/woocommerce/woocommerce/discussions/42996#discussioncomment-8362563 is the main blocker for this. While its true removing unnecessary fields like phone number makes a lot of sense, billing address is still required for payment methods and to determine customer location for things such as tax. The default has to work with these systems. 

If you're suggesting this can be instead done by the payment system, one idea could be for payment methods to flag whether for not they require a billing address so fields can be hidden, but I'd like to a) see examples of payment methods which allow this and b) like to know how taxes for example could be synced back to woo if thats being done after checkout. I suspect taxes would be the most problematic case.

Or there are express checkout methods which bypass WooCommerce fields completely.

**Reply by PlatonB** [reply]
Despite the tremendous sophistication of Western countries, for some reason paper bureaucracy is still prevalent there. Since the development of Woo is based in the West, it is logical that the devs find it hard to imagine that in other parts of the world, commerce can be automated and debureaucratized. As a small digression, in Moscow you can pay for the subway by eyes and receive orders by wheeled robots, while in China delivery by drones is becoming increasingly popular. There are also “free economic zones” in the world, where companies are fully or partially exempt from taxes. Returning to the discussion about checkout fields, I will use the example of the post-Soviet space that a huge questionnaire has two kinds of impact on a potential customer: 1. a person will be too lazy to fill in a lot of fields; 2. a person will be afraid that a ton of advertising will come to the mailing address, spam will come to e-mail, and advertisers and scammers will start calling on the phone around the clock (yes, it's a reality:( ). Therefore, in post-Soviet online stores, the buyer is often asked only one type of communication with the seller, and if possible is not required an address. Also very often the sales process is carried out bypassing the online store in communication via Telegram, Whatsapp or Direct. To summarize, I would like to say that hardcoding checkout fields based only on the EU experience is a bad practice. A seller should be able to customize the fields themselves, striking a balance between ease of checkout and compliance with the laws of their country.

**Reply by senadir** [reply]
There isn't anything actionable or data-reinforced in your comment, this is, and will be, anecdotes. We tend to optimize to the majority of our users but leave the tools and ability to change for everyone, that's the power of open source, the code is just there.

> the development of Woo is based in the West

That's not entirely true, the company Automattic (behind Woo) is based on 98 countries around the world, just the team working on Checkout is based from different countries including north africa, south east asia, and central europe.

Still, the code shared below should give you the ability to modify whatever you want. We won't ship a UI change that might cater to a single digit % of our merchants while making it easy to mess up for a large % of them.

**Reply by PlatonB** [reply]
> There isn't anything actionable or data-reinforced in your comment

Everything I have said above is based on my own life experience. But if you want a nerdy set of proofs, you can explore a [large list of examples](https://tilda.cc/ru/madeontilda/#c=store) of successful post-Soviet stores made on the most popular website builder in this part of the planet.

Here's a [random example](https://highfit.by/tproduct/538083507151-velosipedki) of a Belarusian store that doesn't ask for a shipping address, even though the items are physical.

A [more complex example](https://assoro.ru/catalog/silk-clothes/tproduct/882422223851-top-shelkovii-chernii): the address is only requested when the buyer has chosen home delivery. If the customer has selected a postal service pick-up point or direct store pick-up, a home address is not required.

> Still, the code shared below should give you the ability to modify whatever you want. We won't ship a UI change that might cater to a single digit % of our merchants while making it easy to mess up for a large % of them.

Thanks for the code, but wouldn't it be better to implement a checkbox/toggle element against each checkout field? It would be closer to Gutenberg philosophy (“without any coding knowledge”).

**Reply by senadir** [reply]
> Everything I have said above is based on my own life experience. But if you want a nerdy set of proofs, you can explore a [large list of examples](https://tilda.cc/ru/madeontilda/#c=store) of successful post-Soviet stores made on the most popular website builder in this part of the planet.

It's not about being nerdy, WooCommerce serves millions of stores, unless something is statically significant, we can't add it to the UI for everyone.

I also run an online store in Algeria, a place in which people don't have email, orders are linked to a phone number, and 90% of orders are going toward pickup locations, an address is only collected if you ask for home delivery.

I understand the frustration of felling like your case isn't covered, your use case is similar to mine and I couldn't fully achieve it with the existing extensibility methods, let alone the UI.

Yet the reality is that unless we have a significant evidence that those things are popular, we can't really go and build it vs other things we have.

> Thanks for the code, but wouldn't it be better to implement a checkbox/toggle element against each checkout field? It would be closer to Gutenberg philosophy (“without any coding knowledge”).

Unless we feel confident that disabling a field won't block Checkout, we won't implement it yet, this is what we're working toward. If none of your installed plugins need said field, you would be able to disable it in the editor.

**Reply by PlatonB** [reply]
Why do you think that request for ability to remove checkout fields is not popular? Were the corresponding statistics obtained? If yes, by which method?

**Comment by senadir** [comment]
As Mike said, this is a tricky problem to solve from our side mainly due how payment and shipping plugins expects things to work, and how each country has its own system.

Use the following code at your risk, but I suggest you test it with each country you sell to with your payment methods:


```php
add_filter('woocommerce_get_country_locale', function( $locale ) {
		foreach ( $locale as $key => $value ) {
			$locale[ $key ]['address_1'] = [
				'required' => false,
				'hidden'   => true,
			];

			$locale[ $key ]['postcode'] = [
				'required' => false,
				'hidden'   => true,
			];

			$locale[ $key ]['city'] = [
				'required' => false,
				'hidden'   => true,
			];

			$locale[ $key ]['state'] = [
				'required' => false,
				'hidden'   => true,
			];
		}

		return $locale;
});
```
